### PR TITLE
Display exception cause if possible

### LINKED ISSUE #1768: Improve error messages
When we get an error, we see the less than helpful `Failed to execute the build`.

I think functions like buildAndTag() need to display the underlying cause :

`
throw new MojoExecutionException("Failed to execute the build: " + ex.getCause());
`

### PR DESCRIPTION
**Pull request by plord12**
#1768

### CODE REVIEW

File: plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/BuildMojo.java [hunk 1]
```diff
@@ -296,7 +296,11 @@ protected void buildAndTag(ServiceHub hub, ImageConfiguration imageConfig)
             fabric8ServiceHub.getBuildService().build(imageConfig);
 
         } catch (Exception ex) {
-            throw new MojoExecutionException("Failed to execute the build: " + ex);
+            if (ex.getCause() != null) {
```
Comment (manusa) [inline plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/BuildMojo.java]: nit: more readable one-liner:
```
throw new MojoExecutionException("Failed to execute the build: " + 
Optional.ofNullable(ex.getCause()).orElse(ex));
```
```

**Comment by rohanKanojia** [comment]
@plord12 : Could you please add a line to CHANGELOG.md regarding this change?

**Comment by plord12** [comment]
> @plord12 : Could you please add a line to CHANGELOG.md regarding this change?

I tried but github wouldn't create the pull request.

**Comment by manusa** [comment]
@rohanKanojia Yep, sorry my mailbox got flooded (again) and this one got lost

**Pull request merged**